elastic/kibana
# [Dashboard First] Library Notification

## Summary

Closes #75499

### Reasoning
Adds a notification which denotes when an embeddable is saved to the Visualize Library. Once the 'by value' paradigm is an accepted default, it will be important to be able to distinguish between 'by value' and 'by reference' embeddables at a glance, due to the fact that editing by reference embeddables can apply your changes to multiple dashboards.

This notification shows up when:

- The Embeddable implements `ReferenceOrValueEmbeddable` i.e. is capable of taking input that contains either a savedObjectId OR the saved object attributes.
- Is currently using 'by reference' type input.
- Is on a container in edit mode.

### Implementation
The Library notification is an EUI Badge using the hollow color scheme to make it stand out less. This badge could start showing up in a lot of places once the 'allowByValueEmbeddables' flag is turned on, so it's important to make it less conspicuous.

The custom badge is implemented via the MenuItem prop on action, which I have changed the notifications area of embeddable panel header to accept.

### Screenshots

The Library notification on Book embeddables...
<img width="950" alt="Screen Shot 2020-08-27 at 10 51 24 AM" src="https://user-images.githubusercontent.com/14276393/91473204-89893400-e866-11ea-8b65-2d170620e572.png">

The Library notification appearing on a Lens By Value https://github.com/elastic/kibana/pull/70272 embeddable. The Drilldown notification and local time range are shown for context.

 <img width="1255" alt="Screen Shot 2020-08-19 at 4 09 24 PM" src="https://user-images.githubusercontent.com/14276393/90690359-904cf100-e23f-11ea-9cb6-db5cda718d95.png">

## How to test this?

This is just one part of a much larger change, so all changes in this PR are hidden behind a configuration value.

1. you'll need to set the `allowByValueEmbeddables` config value to `true`:
https://github.com/elastic/kibana/blob/5b64a4cdfea92639ddf9e23c86c43dda5ea3ac68/src/plugins/dashboard/config.ts#L23

2. There are no production ready embeddables that implement the `ReferenceOrValueEmbeddable` yet so you will have to start the developer examples. `yarn start --run-examples`

3. Open a dashboard, and use the add panels functionality to add a Book embeddable. You should see the new notification show up on the right side.

4. Use the 'Unlink from Library' functionality to unlink the panel, and see that the notification disappears.


### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
